closed/custom/Docs.gmk

```diff
@@ -116,3 +116,174 @@ $(OPENJ9_JAVADOC_MARKER) : $(OPENJDK_JAVADOC_MARKER)
 			| $(XARGS) $(BASH) $(OPENJ9_JAVADOC_SCRIPT)
 	$(RM) -f $(OPENJ9_JAVADOC_SCRIPT)
 	$(TOUCH) $@
+
+##############################################################################
+# Create javadoc for only the OpenJ9 extensions and modifications.
+#
+# javadoc automatically retrieves module and package descriptions from the
+# module-info.java, package-info.java and package.html files and uses the
+# information as part of the module-summary.html and package-summary.html
+# files. The module-info.java, # package-info.java and package.html files
+# used would be from the openjdk project source files if they were not
+# overridden by openj9.
+# The text taken from these files is replaced by creating a package-info.java
+# stub file for each package without a package-info.java file provided by
+# openj9, or by removing the text directly from the generated
+# module-summary.html file.
+
+JAVASE_BASE_URL := https://docs.oracle.com/en/java/javase/$(VERSION_SPECIFICATION)/docs/api
+
+OPENJ9_ONLY_JAVADOC_MARKER := $(SUPPORT_OUTPUTDIR)/openj9-docs/_javadoc_openj9_only_exec.marker
+OPENJ9_ONLY_JAVADOC_SCRIPT := $(SUPPORT_OUTPUTDIR)/openj9-docs/_javadoc_openj9_only_replace.sh
+OPENJ9_ONLY_JAVADOC_TEMP  := $(SUPPORT_OUTPUTDIR)/openj9-docs/_javadoc_temp.html
+JDK_API_CUSTOM_TARGETS += $(OPENJ9_ONLY_JAVADOC_MARKER)
+
+OPENJ9_ONLY_JDK_NAME := OpenJ9 JDK $(VERSION_SPECIFICATION)
+OPENJ9_ONLY_DOCDIR := $(IMAGES_OUTPUTDIR)/openj9-docs/api
+OPENJ9_ONLY_INDEX_HTML := $(IMAGES_OUTPUTDIR)/openj9-docs/api/index.html
+
+# Create a list of the openj9 source files for which javadoc is required.
+# These are the openj9.xxxx packages and the jdk management packages with
+# public methods which are openj9 specific extensions or which modify the
+# openjdk implementations behaviour.  javadoc is not generated for the
+# classes  which are replaced by openj9 but are functionally identical
+# to the openjdk implmentations (e.g. java.lang.String), since the openjdk
+# javadoc is still applicable.
+# 
+# The list if source files is passed explicitly (via OPENJ9_ONLY_FILES)
+# to the javadoc tool on the command line and used as dependencies
+# (via OPENJ9_ONLY_FILELIST) to identify when the javadoc needs regenerating.
+
+OPENJ9_ONLY_FILES := $(SUPPORT_OUTPUTDIR)/openj9-docs/_javadoc_openj9_files.txt
+CREATE_FILELIST_RESULT := $(shell $(MKDIR) -p $(SUPPORT_OUTPUTDIR)/openj9-docs ; $(FIND) $(SUPPORT_OUTPUTDIR)/j9jcl_sources -type f -path '*/openj9.*/*.java' -o -path '*/jdk.management/*.java' >$(OPENJ9_ONLY_FILES))
+ifneq ($(CREATE_FILELIST_RESULT), )
+  $(error closed/custom/Docs.gmk: Error creating $(OPENJ9_ONLY_FILES): $(CREATE_FILELIST_RESULT))
```

Comment:
The reference to `closed/custom/Docs.gmk` is redundant: make will say where the error occurred.

Reply:
Message removed as a result of creating the file inside the build target.